### PR TITLE
Add checks for ddo chain id and did. 

### PR DESCRIPTION
**Pull request by mariacarmina**
Fixes # .

Changes proposed in this PR:

- Add checks for ddo chain id and did. 
- Added check for confidential evm.

### CODE REVIEW

**Comment by jamiehewitt15** [comment]
Can you check what thee issue with the integration test is?

**Comment by mariacarmina** [comment]
> Can you check what the issue with the integration test is?

It seems that the modifications work fine :)) so the tests failed as expected, I'll modify the chain id from the samples DDOs to match barge test chain id.

File: src/config/ConfigHelper.ts [hunk 1]
```diff
@@ -339,7 +339,9 @@ export class ConfigHelper {
       }
     }
 
-    config.accessListFactory = contractAddressesConfig.accessListFactory
```
Comment (paulo-ocean) [inline src/config/ConfigHelper.ts:342]: Why this change? It makes no difference, except that now we have more lines of code :-)

Comment (mariacarmina) [reply src/config/ConfigHelper.ts:342]: I saw some errors when testing vscode extension and importing the latest version of the lib, that accessListFactory key is undefined from contractAddressesConfig, because it was mandatory and did not have the check before.

Comment (paulo-ocean) [reply src/config/ConfigHelper.ts:342]: but `accessListFactory` is not mandatory, this is how it is defined on Config class:
`accessListFactory?: string`
And the changes does nothing

**Review by paulo-ocean — APPROVED** [review]
lgtm
